### DISCUSSION TITLE
unable to connect to TM1py with Nginx https non standard url

### DISCUSSION THREAD

**Discussion by SatyaDnv** [cubewise-code/tm1py#743] [General]
The standard URL has https://servername:port/api/v1.. for us the onpremise uses nginx setup and has the https://servername:port/tm1/test/

The connection is refused with 404 or 401 as the default behaviour points to api/v1 ... 

how to change the default behaviour and use /tm1/test in restservice.py,etc...

**Comment by MariusWirtz** [comment]
Hi @SatyaDnv,

does it work if you provide the base_url parameter instead:

``` python
from TM1py import TM1Service

with TM1Service(base_url="https://localhost:12354", user="admin", password="apple") as tm1:
    print(tm1.server.get_product_version())

```

**Comment by SatyaDnv** [comment]
Thank you for the response. 

No. It sends 404 not found exception.

**Comment by SatyaDnv** [comment]
ERROR:
        Text: '<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.18.0</center>
</body>
</html>
' - Status Code: 404 - Reason: 'Not Found' - Headers: {'Server': 'nginx/1.18.0', 'Date': 'Wed, 08 Jun 2022 15:12:11 GMT', 'Content-Type': 'text/html', 'Content-Length': '153', 'Connection': 'keep-alive'}

--

**Comment by MariusWirtz** [comment]
what request do you do against nginx to get the $metadata document in the browser?

Without a proxy you would do this:
`https//localhost:12354/api/v1/$metadata`

**Comment by SatyaDnv** [comment]
The following request response is working well. however, TM1py is 404 not found error.

sConnection_String = 'https://servername:8001/tm1/dev/dashboards'
sCube_Name = 'detailed'
sPrivate_View_Name = 'reportview'
sRequest = sConnection_String + "/Cubes('" + sCube_Name + "')" \
           "/Views('" + sPrivate_View_Name + "') /tm1.Execute?" \
           "$expand=Axes($expand=Hierarchies($select=Name),Tuples($expand=Members($select=Name))),Cells($select=Ordinal,Value)"
response = [requests.post](http://requests.post/)(sRequest,
                         json='Content-type: application/json; charset=utf-8',
                         auth=HttpNegotiateAuth()
                        )
print(response.json())

**Comment by SatyaDnv** [comment]
https//server0:12354/api/v1/$metadata gaves me response of xml of metadata

**Comment by MariusWirtz** [comment]
I think the only solution is to re-map the delegation of your Nginx.
Instead of delegating requests to `https://localhost:12354/api/v1/...` it needs to delegate to `https://localhost:12354/...`

**Comment by SatyaDnv** [comment]
unfortunately, can't change on the system side. Any alternative suggestion on client side is feasible for us.

**Comment by SatyaDnv** [comment]
With 741. i recieve following error.

ERROR:
        No response returned from URL: 'https://servername:8001/tm1/dev/dashboards/api/v1/Configuration/ProductVersion/$value'. Please double check your address and port number in the URL.
--

**Comment by SatyaDnv** [comment]
thank you so much for all your support. The branch of 741 helped us.
